benchmarks/tasks.py

```diff
@@ -39,32 +78,33 @@ def set_testjob_results(self, testjob):
     if not testjob.initialized:
         testjob.testrunnerclass = tester.get_result_class_name(testjob.id)
         testjob.initialized = True
-        testjob.save()
         tester = getattr(testminer, testjob.testrunnerclass)(
             testjob.testrunnerurl, username, password
         )
 
     if testjob.status not in ["Complete", "Incomplete", "Canceled"]:
-        logger.debug("Saving job({0}) status: {1}".format(testjob.id, testjob.status))
-        testjob.save()
+        logger.debug("Job({0}) status: {1}".format(testjob.id, testjob.status))
         return
 
     testjob.definition = tester.get_test_job_details(testjob.id)['definition']
     testjob.completed = True
     logger.debug("Test job({0}) completed: {1}".format(testjob.id, testjob.completed))
     if testjob.status in ["Incomplete", "Canceled"]:
-        logger.debug("Saving job({0}) status: {1}".format(testjob.id, testjob.status))
-        testjob.save()
+        logger.debug("Job({0}) status: {1}".format(testjob.id, testjob.status))
         return
 
     logger.debug("Calling testminer")
     logger.debug("Tester class:{0}".format(tester.__class__.__name__))
     logger.debug("Testjob:{0}".format(testjob.id))
+
+    # already got test results; bail out
+    if testjob.result.data.count() > 0:
```

Comment:
Looks almost OK. I'm not really sure what happens if there are 2 testjobs contributing to the same Result object. Will try it locally.